File: 2215-find-the-difference-of-two-arrays/2215-find-the-difference-of-two-arrays.py
class Solution(object):
    def findDifference(self, nums1, nums2):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: List[List[int]]
        """
        set1, set2 = set(nums1), set(nums2)

        distinct1 = list(set1 - set2)
        distinct2 = list(set2 - set1)

        return [distinct1, distinct2]
        
        

File: 2215-find-the-difference-of-two-arrays/2215-find-the-difference-of-two-arrays.rs
use std::collections::HashSet;

impl Solution {
    pub fn find_difference(nums1: Vec<i32>, nums2: Vec<i32>) -> Vec<Vec<i32>> {
        let set1: HashSet<i32> = nums1.into_iter().collect();
        let set2: HashSet<i32> = nums2.into_iter().collect();

        let distinct1: Vec<i32> = set1.difference(&set2).cloned().collect();
        let distinct2: Vec<i32> = set2.difference(&set1).cloned().collect();

        vec![distinct1, distinct2]
    }
}